test/preset-wind4.test.ts

```diff
@@ -44,6 +44,9 @@ describe('preset-wind4', () => {
         "color-bluegray-400/10",
         "color-bluegray/10",
         "text-red2",
+        "text-[color:--variable]",
+        "text-[color:var(--color)]",
+        "text-[color:var(--color-x)]:[trick]",
```

Comment:
Due to the use of `getStringComponents` to parse the `separator`, it is difficult to support the format of `color:xxx`.